### PR TITLE
test/object_store: Generalize test table creation

### PR DESCRIPTION
**Pull request by xemul**
All two and the upcoming third test cases in the test create the very same ks.cf pair with the very same sequence of steps. Generalize them.

For the basic test case also tune up the way "expected" rows are calculated -- now they are SELECT-ed right after insertion and the size is checked to be non zero. Not _exactly_ the same check, but it's good enough for basic testing purposes.

### CODE REVIEW

File: test/object_store/test_basic.py [hunk 1]
```diff
@@ -109,31 +109,42 @@ def managed_cluster(run_dir, ssl, s3_server):
...
+    cql.execute(f"CREATE TABLE {ks}.{cf} ( name text primary key, value text );")
+
     rows = [('0', 'zero'),
             ('1', 'one'),
             ('2', 'two')]
+    for row in rows:
+        cql_fmt = "INSERT INTO {}.{} ( name, value ) VALUES ('{}', '{}');"
+        cql.execute(cql_fmt.format(ks, cf, *row))
+
+    return (ks, cf)
```
Comment (tchaikov) [inline test/object_store/test_basic.py:133]:
```suggestion
    return ks, cf
```

File: test/object_store/test_basic.py [hunk 2]
```diff
@@ -109,31 +109,42 @@ def managed_cluster(run_dir, ssl, s3_server):
...
-                                     bucket=s3_server.bucket_name)
-
         conn = cluster.cql.connect()
-        conn.execute((f"CREATE KEYSPACE {ks} WITH"
-                      f" REPLICATION = {replication_opts} AND STORAGE = {storage_opts};"))
-        conn.execute(f"CREATE TABLE {ks}.{cf} ( name text primary key, value text );")
-        for row in rows:
-            cql_fmt = "INSERT INTO {}.{} ( name, value ) VALUES ('{}', '{}');"
-            conn.execute(cql_fmt.format(ks, cf, *row))
+        (ks, cf) = create_ks_and_cf(conn, s3_server)
```
Comment (tchaikov) [inline test/object_store/test_basic.py:143]:
```suggestion
        ks, cf = create_ks_and_cf(conn, s3_server)
```

File: test/object_store/test_basic.py [hunk 3]
```diff
@@ -109,31 +109,42 @@ def managed_cluster(run_dir, ssl, s3_server):
...
+
         res = conn.execute(f"SELECT * FROM {ks}.{cf};")
+        rows = {x.name: x.value for x in res}
+        assert len(rows) > 0, f'Test table is empty'
```
Comment (tchaikov) [inline test/object_store/test_basic.py:147]:
```suggestion
        assert len(rows) > 0, 'Test table is empty'
```

File: test/object_store/test_basic.py [hunk 4]
```diff
@@ -183,19 +189,8 @@ def list_bucket(s3_server):
...
-                                     bucket=s3_server.bucket_name)
-
         conn = cluster.cql.connect()
-        conn.execute((f"CREATE KEYSPACE {ks} WITH"
-                      f" REPLICATION = {replication_opts} AND STORAGE = {storage_opts};"))
-        conn.execute(f"CREATE TABLE {ks}.{cf} ( name text primary key, value text );")
-        for row in rows:
-            cql_fmt = "INSERT INTO {}.{} ( name, value ) VALUES ('{}', '{}');"
-            conn.execute(cql_fmt.format(ks, cf, *row))
+        (ks, cf) = create_ks_and_cf(conn, s3_server)
```
Comment (tchaikov) [inline test/object_store/test_basic.py:193]:
```suggestion
        ks, cf = create_ks_and_cf(conn, s3_server)
```

**Review by tchaikov — APPROVED** [review]
lgtm. left some comments on nits.

**Comment by tchaikov** [comment]
> * [test_simple_add_new_node_while_schema_changes_with_stream](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/4590/testReport/junit/update_cluster_layout_tests/TestUpdateClusterLayout/Tests___Sanity_Tests___test_simple_add_new_node_while_schema_changes_with_stream) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_simple_add_new_node_while_schema_changes_with_stream)

new flaky issue. filed #15990 15990

**Comment by xemul** [comment]
@tchaikov , thank you for the feedback. All comments fixed

**Comment by tchaikov** [comment]
@xemul hi Pavel, did you forget to repush the updated branch? i cannot find the delta between the latest revision and the one i reviewed.

**Comment by xemul** [comment]
> @xemul hi Pavel, did you forget to repush the updated branch? i cannot find the delta between the latest revision and the one i reviewed.

I forgot to amend the changes and pushed only rebase part :flushed: Fixed

**Review by tchaikov — APPROVED** [review]
lgtm

**Comment by xemul** [comment]
#16005

**Review by denesb — APPROVED** [review]

**Pull request closed without merging**